### PR TITLE
Implement `PrivateIdentifier in ...` expression

### PR DESCRIPTION
**Pull request by ksh8281**
Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

**Review by bbrto21 — APPROVED** [review]

File: src/runtime/Object.cpp [hunk 1]
```diff
@@ -1943,6 +1943,18 @@ Value Object::getPrivateMember(ExecutionState& state, Object* contextObject, Ato
...
+bool Object::hasPrivateMember(ExecutionState& state, Object* contextObject, AtomicString propertyName, bool shouldReferOuterClass)
+{
+    auto e = ensureExtendedExtraData();
+    auto piece = findPieceOnPrivateMemberChain(state, e, contextObject);
+    if (piece) {
+        return piece->m_privateMemberStructure->findProperty(propertyName);
+    } else if (shouldReferOuterClass && contextObject->asScriptClassConstructorFunctionObject()->outerClassConstructor()) {
+        return hasPrivateMember(state, contextObject->asScriptClassConstructorFunctionObject()->outerClassConstructor().value(), propertyName);
+    }
+    return false;
```
Comment (zherczeg) [inline src/runtime/Object.cpp:1955]: There are two returns above this return, so I think this is dead code.

**Review by zherczeg — COMMENTED** [review]
Nice patch! I have only one question.

File: src/runtime/Object.cpp (hunk 1 above)
Comment (ksh8281) [reply src/runtime/Object.cpp:1955]: I think `!piece && !(shouldReferOuterClass && contextObject->asScriptClassConstructorFunctionObject()->outerClassConstructor())` could be exists

Comment (zherczeg) [reply src/runtime/Object.cpp:1955]: You are right. Sorry.

Comment (ksh8281) [reply src/runtime/Object.cpp:1955]: I am fine :)

Comment (clover2123) [reply src/runtime/Object.cpp:1955]: I have a question about `shouldReferOuterClass` parameter.
Do we have to check outer class too?
And `shouldReferOuterClass` parameter is enabled by default, but actually called as false below

```c++
registerFile[code->m_dstIndex] = Value(right.toObject(state)->hasPrivateMember(state, state.findPrivateMemberContextObject(), AtomicString(state, left.asString()), false));
```

Comment (ksh8281) [reply src/runtime/Object.cpp:1955]: Currently, there is no such case(shouldReferOuterClass is false)
but I added it for future && private member variable get operations could refer outer class

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**